set_test.go

```diff
@@ -171,3 +173,59 @@ func TestRegisterUnregister(t *testing.T) {
 	}
 	wg.Wait()
 }
+
+func TestExposeMetadata(t *testing.T) {
```

Comment:
I'd recommend tweaking `ExampleExposeMetadata()` test instead of adding these new tests. It can contain all cases:
* metrics that weren't updated
* metrics that were sorted in a way that quantiles go first

Reply:
I see. thanks for catching this, it was missed before.

Now the sorting is rewritten and test cases are merged to `ExampleExposeMetadata()`. PTAL.

Note that I also exchange the `metricType` for `summary` (count and sum) and `quantiles`.